lightkurve/io/detect.py

```diff
@@ -36,11 +37,17 @@ def detect_filetype(hdulist: HDUList) -> str:
         A string describing the detected filetype. If the filetype is not
         recognized, `None` will be returned.
     """
+
     # Is it a MIT/QLP TESS FFI Quicklook Pipeline light curve?
     # cf. http://archive.stsci.edu/hlsp/qlp
     if "mit/qlp" in hdulist[0].header.get('origin', '').lower():
         return "QLP"
 
+    # Is it a PATHOS TESS light curve?
+    # cf. http://archive.stsci.edu/hlsp/pathos
+    if "pathos" in (hdulist.filename() or ""):
```

Comment:
Oops, it looks like we can't rely on `pathos` appearing in the filename when a file is opened using astropy's `fits.open()`, because astropy renames the file using a hash.  For example, `hdulist.filename()` is equal to `"/Users/gb/.astropy/cache/download/url/bff14286987df5550091e2ad8c8dae9b/contents"` when I run the `test_detect_pathos` test on my system. 

This makes life difficult because there are no obvious header keywords in PATHOS light curves that we can use to detect the product.  My best suggestion is to identify the product based on the fingerprint of column names?  For example, we could try to detect the product as follows:
```suggestion
    if all(x in f[1].columns.names for x in ["PSF_FLUX_RAW", "PSF_FLUX_COR", "AP4_FLUX_RAW", "AP4_FLUX_COR", "SKY_LOCAL"]):
```

We'd have to cross our little fingers that no other product will ever use the same combination of column names, but it's the only solution I can think of.

(If we do it this way, we'll want to explain this problem in a little comment.)